wrong way to calculate the amount of items in apiInfo.ldapai_extensions

GCC gives a warning:
In file included from /home/pkg-dev/tmp/build/QC_LdapClient.cpp:26:
/home/pkg-dev/git/qore-openldap/src/QoreLdapClient.h: In destructor 'QoreLDAPAPIInfoHelper::~QoreLDAPAPIInfoHelper()':
/home/pkg-dev/git/qore-openldap/src/QoreLdapClient.h:359:72: warning: division 'sizeof (char**) / sizeof (char*)' does not compute the number of array elements [-Wsizeof-pointer-div]
  359 |                 size_t nextensions = sizeof(apiInfo.ldapai_extensions) / sizeof(apiInfo.ldapai_extensions[0]);
      |                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Running the testprogram below:
#include <iostream>
#include <ldap.h>

int
main()
{
  LDAPAPIInfo apiInfo;
  char array[123];
  apiInfo.ldapai_info_version = LDAP_API_INFO_VERSION;  
  int ec = ldap_get_option(0, LDAP_OPT_API_INFO, &apiInfo);

  size_t s_array = sizeof(array);
  size_t api_ext = sizeof(apiInfo.ldapai_extensions);
  size_t api_ext_item = sizeof(apiInfo.ldapai_extensions[0]);
 
  std::cout << "api_ext: " << api_ext << std::endl;
  std::cout << "api_ext_item: " << api_ext_item << std::endl;
  std::cout << "array: " << s_array << std::endl;
  
  return 0;
}

gives the output:
api_ext: 8
api_ext_item: 8
array: 123

and that confirms GCCs warning the calculation sizeof(apiInfo.ldapai_extensions) / sizeof(apiInfo.ldapai_extensions[0]);
will just return 1 even if there are more items, which leads to them not being freed.

--- a/src/QoreLdapClient.h
+++ b/src/QoreLdapClient.h
@@ -356,8 +356,7 @@ public:
             ldap_memfree(apiInfo.ldapai_vendor_name);
             apiInfo.ldapai_vendor_name = 0;
             if (apiInfo.ldapai_extensions) {
-                size_t nextensions = sizeof(apiInfo.ldapai_extensions) / sizeof(apiInfo.ldapai_extensions[0]);
-                for (size_t i = 0; i < nextensions; ++i) {
+                for (size_t i = 0; apiInfo.ldapai_extensions[i]; ++i) {
                     ldap_memfree(apiInfo.ldapai_extensions[i]);
                 }
                 ldap_memfree(apiInfo.ldapai_extensions);

